bindings/rust/evmc-vm/src/lib.rs

```diff
@@ -2,6 +2,77 @@ pub extern crate evmc_sys;
 pub use evmc_sys as ffi;
 
 // TODO: Add helpers for host interface
-// TODO: Add convenient helpers for evmc_result
 // TODO: Add convenient helpers for evmc_execute
 // TODO: Add a derive macro here for creating evmc_create
+
+/// EVMC result structure.
+pub struct EvmcResult(Box<ffi::evmc_result>);
+
+impl EvmcResult {
+    /// Initialize a new result struct given a status code, remaining gas, and optional output
+    /// data.
+    pub fn new(status: ffi::evmc_status_code, gasleft: i64, output: Option<&[u8]>) -> Self {
+        EvmcResult(Box::new(ffi::evmc_result {
+            status_code: status,
+            gas_left: gasleft,
+            output_data: if let Some(buf) = output {
+                buf.clone().as_ptr()
+            } else {
+                std::ptr::null()
+            },
+            output_size: if let Some(buf) = output { buf.len() } else { 0 },
+            release: Some(release_result),
+            create_address: ffi::evmc_address { bytes: [0u8; 20] },
+            padding: [0u8; 4],
+        }))
+    }
+
+    pub fn gasleft(&self) -> i64 {
+        self.0.gas_left
+    }
+
+    pub fn create_addr(&self) -> &ffi::evmc_address {
+        &self.0.create_address
+    }
+
+    pub fn output(&self) -> Option<&[u8]> {
+        if !self.0.output_data.is_null() && self.0.output_size != 0 {
+            Some(unsafe {
+                std::slice::from_raw_parts::<u8>(self.0.output_data, self.0.output_size)
+            })
+        } else {
+            None
+        }
+    }
+
+    pub fn status(&self) -> ffi::evmc_status_code {
+        self.0.status_code
+    }
+}
+
+#[no_mangle]
```

Comment:
I don't think the `no_mangle` is needed since we pass this around as a function pointer.